securesystemslib/signer/_key.py

```diff
@@ -0,0 +1,207 @@
+"""Key interface and the default implementations"""
+import abc
+import logging
+from typing import Any, Dict, Optional, Tuple, Type
+
+import securesystemslib.keys as sslib_keys
+from securesystemslib import exceptions
+from securesystemslib.signer._signature import Signature
+
+logger = logging.getLogger(__name__)
+
+# NOTE dict for Key dispatch defined here, but filled at end of file when
+# subclass definitions are available. Users can add Key implementations.
+
+KEY_FOR_TYPE_AND_SCHEME: Dict[Tuple[str, str], Type] = {}
+
+
+class Key:
+    """Abstract class representing the public portion of a key.
+
+    *All parameters named below are not just constructor arguments but also
+    instance attributes.*
+
+    Args:
+        keyid: Key identifier that is unique within the metadata it is used in.
+            Keyid is not verified to be the hash of a specific representation
+            of the key.
+        keytype: Key type, e.g. "rsa", "ed25519" or "ecdsa-sha2-nistp256".
+        scheme: Signature scheme. For example:
+            "rsassa-pss-sha256", "ed25519", and "ecdsa-sha2-nistp256".
+        keyval: Opaque key content
+        unrecognized_fields: Dictionary of all attributes that are not managed
+            by Securesystemslib
+
+    Raises:
+        TypeError: Invalid type for an argument.
+    """
+
+    __metaclass__ = abc.ABCMeta
```

Comment:
this apparently does not do the same things as deriving from ABC: calling Key() should fail now as Key methods are not all implemented, but it does not. If Key derives from ABC, constructor starts failing as expected

Reply:
Nice find here. I'm genuinely surprised it took us so long to notice... I opened a draft PR in #465 (need to test it with some sslib dependents before it's ready for merge).

Reply:
Ok, I'll probably try to fix this one here then.